Gemfile

```diff
@@ -24,13 +24,22 @@ if File.exist?(file)
 else
   Bundler.ui.warn "[EngineCart] Unable to find test application dependencies in #{file}, using placeholder dependencies"
 
-  gem 'rails', ENV['RAILS_VERSION'] if ENV['RAILS_VERSION']
+  if ENV['RAILS_VERSION']
```

Comment:
I don't think you need the outer `if` here. If `ENV` doesn't contain the key `RAILS_VERSION`, the `if` on the following line won't blow up; it'll just return `false`.